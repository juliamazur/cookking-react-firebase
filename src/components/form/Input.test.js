import * as React from 'react';
import * as enzyme from 'enzyme';
import Input from './Input';

it('Input container is displayed', () => {
  const tree = enzyme.shallow(<Input/>);
  const container = tree.find({"data-test": 'input'});
  expect(container.length).toEqual(1);
});

describe("Input", () => {
  it('should be defined', () => {
    expect(Input).toBeDefined();
  });
  it('should render correctly', () => {
    const tree = enzyme.shallow(<Input/>);
    expect(tree).toMatchSnapshot();
  });
  it('should render correctly with props', () => {
    const tree = enzyme.shallow(<Input id="test-input" name="Test" value="test-value"/>);
    expect(tree).toMatchSnapshot();
  });
  it('should call mock function on change', () => {
    const mockFn = jest.fn();
    const event = { target: { value: 'the-value' } };
    const tree = enzyme.shallow(<Input handleChange={mockFn}/>);
    const container = tree.find({"data-test": 'input'});
    expect(container.length).toEqual(1);
    container.simulate('change', event);
    expect(mockFn).toHaveBeenCalled();
  });
});