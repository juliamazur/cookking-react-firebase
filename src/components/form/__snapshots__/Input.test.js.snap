// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Input />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <TextField
        data-test="input"
        id={undefined}
        label={undefined}
        margin="normal"
        onChange={undefined}
        placeholder={undefined}
        required={false}
        select={false}
        value={undefined}
        variant="standard"
/>,
      "fullWidth": true,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "data-test": "input",
        "id": undefined,
        "label": undefined,
        "margin": "normal",
        "onChange": undefined,
        "placeholder": undefined,
        "required": false,
        "select": false,
        "value": undefined,
        "variant": "standard",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <TextField
          data-test="input"
          id={undefined}
          label={undefined}
          margin="normal"
          onChange={undefined}
          placeholder={undefined}
          required={false}
          select={false}
          value={undefined}
          variant="standard"
/>,
        "fullWidth": true,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "data-test": "input",
          "id": undefined,
          "label": undefined,
          "margin": "normal",
          "onChange": undefined,
          "placeholder": undefined,
          "required": false,
          "select": false,
          "value": undefined,
          "variant": "standard",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`Input should render correctly with props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Input
    id="test-input"
    name="Test"
    value="test-value"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <TextField
        data-test="input"
        id="test-input"
        label={undefined}
        margin="normal"
        onChange={undefined}
        placeholder={undefined}
        required={false}
        select={false}
        value="test-value"
        variant="standard"
/>,
      "fullWidth": true,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "data-test": "input",
        "id": "test-input",
        "label": undefined,
        "margin": "normal",
        "onChange": undefined,
        "placeholder": undefined,
        "required": false,
        "select": false,
        "value": "test-value",
        "variant": "standard",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <TextField
          data-test="input"
          id="test-input"
          label={undefined}
          margin="normal"
          onChange={undefined}
          placeholder={undefined}
          required={false}
          select={false}
          value="test-value"
          variant="standard"
/>,
        "fullWidth": true,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "data-test": "input",
          "id": "test-input",
          "label": undefined,
          "margin": "normal",
          "onChange": undefined,
          "placeholder": undefined,
          "required": false,
          "select": false,
          "value": "test-value",
          "variant": "standard",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
